p4-16/psa/PSA.mdk

```diff
@@ -3128,3 +3170,192 @@ the performance of these protocols.
 Similarly, there are some protocols like IP header compression, where
 multiple variants have been developed, some with more or less
 robustness in the face of network reordering.
+
+
+# Appendix: Supporting empty action selector groups {#appendix-empty-action-selector-groups}
+
+As mentioned in Section [#sec-action-selector], a PSA data plane
+implementation need not implement any specific defined behavior if one
+attempts to add a table entry for a key that points at a group that is
+currently empty.
+
+Some P4 users have expressed an interest in enabling a P4Runtime
+client (hereafter called the controller) to remove the last member of
+an action selector group, and have the data plane behave in a
+predictable way.
+
+For example, if one has a table that maps logical interface id numbers
+to physical port numbers, using an action selector to implement LAG
+(Link Aggregation Group), what should the controller do when there was
+only one physical port in a LAG that was enabled, and that port goes
+down?  A straightforward desired behavior from the controller's
+perspective is: issue the P4Runtime command to remove the last member
+from the group, and have an empty group action of "drop the packet"
+take effect for any packets applying the table and selecting the empty
+group.
+
+Fully supporting empty action groups should meet all of these
+requirements:
+
+- All P4Runtime API operations such as adding a member to a group
+  (even when changing from empty to 1 member), removing a member from
+  a group (even when changing from 1 member to empty), modifying the
+  action associated with a member, etc. should be atomic relative to
+  data packet processing.  That is, every packet should be processed
+  as if the table was in the old state, or the new state, with no
+  undefined packet processing behavior.
+- The empty group action that is executed when the matching table
+  entry points at an empty group, may have an action name that is the
+  same as, or different than, the action name used by the group when
+  it was non-empty (if it was non-empty, and then became empty by
+  removing its last member).
+
+Achieving all of these requirements seems not to be possible with a
+PSA data plane implementation that meets only the minimum requirements
+for action selectors, i.e. one that restricts members of a group to
+all have the same action name, and that does not natively support
+predictable behavior if a group is empty in the data plane.
+
+Below we describe one way that nearly achieves the goals above, except
+that it requires the empty group action to have the same name as the
+non-empty groups of the action selector.  This may be too onerous of a
```

Comment:
maybe it would be clearer to repeat the word "action" here.

Reply:
If you meant replacing "the same name as" with "the same action name as", then I have made that change.

If  you meant something else, let me know.

Reply:
I may be misunderstanding the sentence, but I meant:
"it requires the empty group action to be the same as the action for non-empty groups"

Reply:
I think we're on the same page, you should feel free to merge
I think replacing "name" with "action name" like you did in your latest commit made the sentence clearer